src/admissioncontrollerpolicy/admissionControllerPolicy.go

```diff
@@ -0,0 +1,294 @@
+			var kyvernoPolicy kyvernov1.Policy
+			err := yaml.Unmarshal(policyYaml.Yaml, &kyvernoPolicy)
+			if err != nil {
+				log.Error().Msgf("unmarshalling policy yaml failed err=%v", err.Error())
+				continue
+			}
+			kyvernoPolicies = append(kyvernoPolicies, kyvernoPolicy)
+		}
+	}
+	return kyvernoPolicies
+}
+
+func ConvertPoliciesToWorkerResponse(policies []kyvernov1.Policy) *wpb.WorkerResponse {
+	var response wpb.WorkerResponse
+
+	for i := range policies {
+		kyvernoPolicy := wpb.Policy{}
+
+		val, err := json.Marshal(&policies[i])
+		if err != nil {
+			log.Error().Msgf("kyvernoPolicy json marshal failed err=%v", err.Error())
+		}
+		kyvernoPolicy.Data = val
+
+		response.AdmissionControllerPolicy = append(response.AdmissionControllerPolicy, &kyvernoPolicy)
+	}
+	response.Res = "OK"
+
+	return &response
+}
+
+func AutoGenPrecondition(templateKey string, labels types.LabelMap, precondition apiextensions.JSON) apiextensions.JSON {
+	preconditionMap := precondition.(map[string]interface{})
+	for key, value := range labels {
+		newPrecondition := map[string]interface{}{
+			"key":      "{{ request.object.spec." + templateKey + ".metadata.labels." + key + " || '' }}",
+			"operator": "Equals",
+			"value":    value,
+		}
+		existingSlice := preconditionMap["all"].([]interface{})
+		preconditionMap["all"] = append(existingSlice, newPrecondition)
+	}
+	return preconditionMap
+}
+
+func AutoGenPattern(templateKey string, pattern apiextensions.JSON) apiextensions.JSON {
+	newPattern := map[string]interface{}{
+		"spec": map[string]interface{}{
+			templateKey: pattern,
+		},
+	}
+	return newPattern
+}
+
+func ShouldSATokenBeAutoMounted(namespace string, labels types.LabelMap) bool {
+	client := cluster.ConnectK8sClient()
+
+	podList, err := client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: libs.LabelMapToString(labels),
+	})
+
+	if err != nil {
+		log.Warn().Msg(err.Error())
+		return true
+	}
+
+	if len(podList.Items) > 0 {
+		// Only inspect the first pod in the list as deployment pods have same behavior
+		pod := podList.Items[0]
+		containersSATokenMountPath, err := getSATokenMountPath(&pod)
+		if err != nil && strings.Contains(err.Error(), "service account token not mounted") {
+			log.Warn().Msg(err.Error())
+			return false
+		}
+		var sumResponses []*opb.Response
+		for _, container := range pod.Spec.Containers {
+			sumResp, err := obs.GetSummaryData(&opb.Request{
+				PodName:       pod.Name,
+				NameSpace:     pod.Namespace,
+				ContainerName: container.Name,
+				Type:          "process,file",
+			})
+			if err != nil {
+				log.Warn().Msg(err.Error())
+				return true
+			}
+			sumResponses = append(sumResponses, sumResp)
+		}
+		return serviceAccountTokenUsed(containersSATokenMountPath, sumResponses)
+	} else {
+		log.Warn().Msg("No pods found for the given labels")
+	}
+
+	return true
+}
+
+func getSATokenMountPath(pod *corev1.Pod) ([]containerMountPathServiceAccountToken, error) {
```

Comment:
What is the point of returning an array of `containerMountPathServiceAccountToken`? Based on the logic of the function there can atmost be one service account token associated with the pod.

Reply:
There can be at most one SA token, but for pod with multiple containers, SA token is mounted on both. Yes, the path is the same, but this model gives more flexibility and returns the mount point of SA token per container and not per pod.